### PR TITLE
Add composite dataset `PartitionedDataSet` class

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

Add `PartitionedDataSet`, which is a composite dataset.

Start of #5377.

### Details

- None

### CODE REVIEW

**Comment by tkoyama010** [comment]
@pyvista-bot preview

**Comment by tkoyama010** [comment]
@pyvista-bot preview

**Comment by tkoyama010** [comment]
@pyvista-bot preview

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

**Comment by tkoyama010** [comment]
@pyvista-bot preview

**Comment by tkoyama010** [comment]
@pyvista-bot preview

**Comment by tkoyama010** [comment]
@pyvista-bot preview

File: doc/source/api/core/partitioned.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
+Partitioned Datasets
+====================
+
+The :class:`pyvista.PartitionedDataSet` class is a partitioned dataset that encapsulates
+a dataset consisting of partitions. ``PartitionedDataSet`` behaves mostly like a list.
+
+List-like Features
```
Comment (untereiner) [inline doc/source/api/core/partitioned.rst:7]: I do not understand the list-like features. For me it's a tree-like structure. Then making it a list is reductive.

File: pyvista/core/partitioned.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+            except:
+                fmt += row.format(attr[0], attr[2].format(attr[1]))
+        fmt += "</table>\n"
+        fmt += "\n"
+        fmt += "</td><td>"
+        fmt += "\n"
+        fmt += "<table>\n"
+        row = "<tr><th>{}</th><th>{}</th><th>{}</th></tr>\n"
+        fmt += row.format("Index", "Name", "Type")
+        for i in range(self.n_blocks):
```
Comment (tkoyama010) [inline pyvista/core/partitioned.py:157]:
```suggestion
        for i in range(self.n_partitions):
```

File: doc/source/api/core/partitioned.rst (hunk 1 above)
Comment (tkoyama010) [reply doc/source/api/core/partitioned.rst:7]: Good point. The tree structure is a `PartitionedDataSetCollection`, which is a collection of `PartitionedDataSets`, and will be added in the next PR after this one.

**Comment by tkoyama010** [comment]
@pyvista-bot preview

**Comment by tkoyama010** [comment]
@pyvista-bot preview

**Comment by tkoyama010** [comment]
@pyvista-bot LGTM

**Comment by tkoyama010** [comment]
@pyvista-bot LGTM

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

File: pyvista/core/_vtk_core.py [hunk 3]
```diff
@@ -427,6 +428,9 @@ def __init__(self):  # pragma: no cover
 with contextlib.suppress(ImportError):
     from vtkmodules.vtkFiltersCore import vtkPackLabels, vtkSurfaceNets3D
 
+with contextlib.suppress(ImportError):
```
Comment (tkoyama010) [inline pyvista/core/_vtk_core.py:431]:
```suggestion
# 9.1+ imports
with contextlib.suppress(ImportError):
```

**Comment by tkoyama010** [comment]
@pyvista-bot LGTM

**Comment by CharlesGueunet** [comment]
Thank you for these awesome improvements :)

**Comment by tkoyama010** [comment]
> Thank you for these awesome improvements :)

We will continue to add the `PartitionedDataSetCollection` class :)

**Pull request merged**